app/connectors/connector-verifier/src/test/java/io/syndesis/verifier/api/SyndesisMetadataTest.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.verifier.api;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Test;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.module.jsonSchema.types.ObjectSchema;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SyndesisMetadataTest {
+
+    @Test
+    public void shouldSerializeWithObjectSchema() throws JsonProcessingException {
+        final Map<String, List<PropertyPair>> properties = new HashMap<>();
+        properties.put("aProperty", Collections.singletonList(new PropertyPair("value", "label")));
+
+        final ObjectSchema inputSchema = new ObjectSchema();
+        inputSchema.setTitle("input");
+
+        final ObjectSchema outputSchema = new ObjectSchema();
+        outputSchema.setTitle("output");
+
+        final SyndesisMetadata<ObjectSchema> syndesisMetadata = new SyndesisMetadata<>(properties, inputSchema, outputSchema);
+
+        final ObjectMapper mapper = new ObjectMapper();
+
+        final String json = mapper.writeValueAsString(syndesisMetadata);
+
+        assertThat(json).isEqualTo(
+            "{\"outputSchema\":{\"type\":\"object\",\"title\":\"output\"},\"inputSchema\":{\"type\":\"object\",\"title\":\"input\"},\"properties\":{\"aProperty\":[{\"displayValue\":\"label\",\"value\":\"value\"}]}}");
```

Comment:
what i've seen to so often now is that JSON documents are compared as strings. This is so fragile (and actually had that issue today already because of accidentally adding whitespace to a json document at the end), we should not do this. No one guarantees you the order of the keys for example and that might change randomly.

Could we use something like jsonassert.skyscreamer.org please ?